src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/ShardedHttpExecutionServiceCluster.scala

```diff
@@ -0,0 +1,115 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import _root_.slick.dbio.Effect.{Write, Read}
+import _root_.slick.dbio._
+import _root_.slick.driver.JdbcDriver
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkflowRecord
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.concurrent._
+import scala.concurrent.Future._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+
+class ShardedHttpExecutionServiceCluster (members: Map[ExecutionServiceId,ExecutionServiceDAO], default: ExecutionServiceId, dataSource: SlickDataSource) extends ExecutionServiceCluster {
+
+  // make a copy of the members map as an array for easy reads
+  private val memberArray:Array[ClusterMember] = (members map {case (id, dao) => ClusterMember(id, dao)}).toArray
+
+  // find the default instance
+  private val defaultEntry:(ExecutionServiceId, ExecutionServiceDAO) =
+    members.find{case (esid, dao) => esid.id.equals(default.id)}
+      .getOrElse(throw new RawlsException(s"cannot initialize ShardedHttpExecutionServiceCluster: member with default key $default does not exist"))
+
+
+  // ====================
+  // facade methods
+  // ====================
+
+  // only used in tests??? TODO: DA update tests to use the same thing runtime does, or remove this.
+  def submitWorkflow(wdl: String, inputs: String, options: Option[String], userInfo: UserInfo): Future[ExecutionServiceStatus] =
+    getDefaultMember.submitWorkflow(wdl, inputs, options, userInfo)
```

Comment:
I'd like to handle submitWorkflow vs. submitWorkflows as a separate PR/story. It's pretty small, but this PR is big enough already.